kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,446 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.nm.configuration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import org.freedesktop.dbus.errors.NotSupported;
+import org.junit.Test;
+
+public class NetworkPropertiesTest {
+
+	NetworkProperties netProps;
+	Map<String, Object> properties = new HashMap<String, Object>();
+	Optional<?> optResult;
+	String stringResult;
+	Boolean booleanResult;
+	Short shortResult;
+	int intResult;
+	Long longResult;
+
+	List<String> stringListResult;
+	Map<String, Object> resultMap;
+
+	Boolean hasNullPointExceptionBeenThrown = false;
+	Boolean hasNoSuchElementExceptionBeenThrown = false;
+
+	@Test
+	public void shouldThrowWithNullMap() {
+		givenNetworkPropertiesBuiltWith(null);
+		thenANullPointerExceptionOccured();
+	}
+
+	@Test
+	public void shouldWorkWithEmptyMap() {
+		givenNetworkPropertiesBuiltWith(new HashMap<String, Object>());
+		whenGetPropertiesIsCalled();
+		thenNoExceptionsOccured();
+		thenMapResultEquals(new HashMap<String, Object>());
+	}
+
+	@Test
+	public void getShouldReturnPropertiesWithGetProperties() {
+		givenMapWith("testKey1", "testString1");
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetPropertiesIsCalled();
+		thenNoExceptionsOccured();
+		thenMapResultEquals(Collections.singletonMap("testKey1", "testString1"));
+	}
+
+	@Test
+	public void getShouldReturnString() {
```

Comment:
What are the testing conditions?

```suggestion
	public void getShouldWorkWithString() {
```